lib/explorer/polars_backend/data_frame.ex

```diff
@@ -129,6 +129,15 @@ defmodule Explorer.PolarsBackend.DataFrame do
     end
   end
 
+  def to_csv_writer_sample(%DataFrame{data: df}, path, header?, delimiter) do
```

Comment:
Please just move this function to after the next clause of `to_csv`. I think this is the reason the tests are breaking.